cluster-location-metadata/README.md

```diff
@@ -0,0 +1,343 @@
+---
+
+# Location metadata for clusters
+
+### Problem statement
+
+We have anecdotal evidence that customers think about our installations in terms of their physical location. They communicate in terms like these (paraphrased):
+
+- the Dublin management cluster
+- the dev cluster in Singapore
+- the China installation
+
+Some on-prem customers use location identifiers like city names in cluster descriptions.
+
+Relevant user stories in this context could be, to name two examples:
+
+- As a platform admin, I want to check the Kubernetes version of all clusters in a certain location.
+- As a service owner, I want to check the deployment status of certain apps in a certain location.
+
+With Cluster API on cloud providers we usually have location indicators in cluster resources, in various places. Each cloud provider uses their own identifier system, however they all make use of a string identifier. The proposed solution in this RFC would help simplify and unify location lookup for clusters by clients throughout cloud providers.
+
+In on-premises installations, no location information is currently available in Kubernetes resources. In metrics (PrometheusMimir) we annotate many series with a `region` label. However, in on-prem installations, the value for this label is always `onprem`. This RFC attempts to change that.
+
+### Key assumptions
+
+This RFC assumes that each cluster exists in one location only, and nodes of the same cluster are not distributed geographically.
+
+### Decision maker
+
+- Team Honeybadger as driver
+- KaaS teams, especially Rocket
+
+### Deadline
+
+None
+
+### Who is affected / stakeholders
+
+- Customers
+- Giant Swarm engineers
+
+### Preferred solution
+
+The solution proposed here consists of two parts:
+
+- Setting consistent metadata on clusters
+- Providing a default location ID system for on-prem customers
+- Setting the `region` label in metrics accordingly
+
+#### Consistent metadata on clusters
+
+We should set the [`topology.kubernetes.io/region`](https://kubernetes.io/docs/reference/labels-annotations-taints/#topologykubernetesioregion) label on the main cluster resource of all clusters. Currently (as of February 2025) this resource is of kind `Cluster`, group=`cluster.x-k8s.io/v1beta1`.
+
+The `topology.kubernetes.io/region` label is documented as w well-known label, set by Kubernetes on `Node` and `PersistentVolume` resources.
+
+For clusters running in cloud providers, the value of the label should be the region identifier. E. g. `eu-west-1` for the AWS region in EU/Dublin.
+
+For clusters on premises, the value should follow a system that we establish for this purpose. The system may differ between customers, to allow fulfilling of customer-specific requirements.
+
+Regardless of the environment (cloud/on-prem), the value of the `topology.kubernetes.io/region` label should not include any whitespace and consist of the lowercase letters, numbers, and dash only (`[a-z0-9-]`).
```

Comment:
Is there a reason why it needs to be lowercase?

Reply:
The goal here is to restrict the character space as much as possible. Cloud providers also use lowercase letters, so our system would resemble the existing identifiers better.

Reply:
I would think label values restriction would bé sufficient but alright

Reply:
looking at https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#syntax-and-character-set, there's also a maximum length

Reply:
I clarified the character restrictions with the latest commit. The length restriction applies to all labels, so I didn't mention that. https://github.com/giantswarm/rfc/pull/112/commits/e73b89cfd869bcad29a27ae977363d5834bcbf97